From 68eb42c009140581130704f4eaeff9dd783cc60b Mon Sep 17 00:00:00 2001
From: sathonay <alexis.chamonix@gmail.com>
Date: Mon, 26 Aug 2019 17:45:53 +0200
Subject: [PATCH] Add owner for EntityItem


diff --git a/src/main/java/net/minecraft/server/EntityHuman.java b/src/main/java/net/minecraft/server/EntityHuman.java
index 7da4467c..2dbd808b 100644
--- a/src/main/java/net/minecraft/server/EntityHuman.java
+++ b/src/main/java/net/minecraft/server/EntityHuman.java
@@ -588,6 +588,7 @@ public abstract class EntityHuman extends EntityLiving implements ICommandListen
             }
             // CraftBukkit end
 
+            entityitem.owner = this;
             this.a(entityitem);
             this.a(StatisticList.s, 1);
             return entityitem;
diff --git a/src/main/java/net/minecraft/server/EntityItem.java b/src/main/java/net/minecraft/server/EntityItem.java
index 97830f35..4bf97f94 100644
--- a/src/main/java/net/minecraft/server/EntityItem.java
+++ b/src/main/java/net/minecraft/server/EntityItem.java
@@ -17,6 +17,7 @@ public class EntityItem extends Entity {
     private String g;
     public float c;
     private int lastTick = MinecraftServer.currentTick; // CraftBukkit
+    public EntityHuman owner;
 
     public EntityItem(World world, double d0, double d1, double d2) {
         super(world);
@@ -59,6 +60,7 @@ public class EntityItem extends Entity {
 
     public void h() {
         if (this.getItemStack() == null) {
+
             this.die();
         } else {
             super.h();
@@ -386,4 +388,9 @@ public class EntityItem extends Entity {
     public void b(String s) {
         this.f = s;
     }
+
+    public void die() {
+        this.owner = null;
+        super.die();
+    }
 }
-- 
2.20.1.windows.1

