From d9ef049c5892f5200b65c67268c50f824ff906ad Mon Sep 17 00:00:00 2001
From: sathonay <sathonaypro@gmail.com>
Date: Fri, 30 Aug 2019 19:52:52 +0200
Subject: [PATCH] set TCP_NODELAY to true


diff --git a/src/main/java/net/minecraft/server/ServerConnectionChannel.java b/src/main/java/net/minecraft/server/ServerConnectionChannel.java
index d7d93a07..adc76c32 100644
--- a/src/main/java/net/minecraft/server/ServerConnectionChannel.java
+++ b/src/main/java/net/minecraft/server/ServerConnectionChannel.java
@@ -22,7 +22,7 @@ class ServerConnectionChannel extends ChannelInitializer {
         }
 
         try {
-            channel.config().setOption(ChannelOption.TCP_NODELAY, Boolean.valueOf(false));
+            channel.config().setOption(ChannelOption.TCP_NODELAY, Boolean.valueOf(true));
         } catch (ChannelException channelexception1) {
             ;
         }
-- 
2.20.1.windows.1

