From 9406c4e8d0eddf6cc44b246776aed2c27768f52c Mon Sep 17 00:00:00 2001
From: sathonay <sathonaypro@gmail.com>
Date: Fri, 30 Aug 2019 21:26:59 +0200
Subject: [PATCH] Make EntityFishingHook respect vanish api


diff --git a/src/main/java/net/minecraft/server/EntityFishingHook.java b/src/main/java/net/minecraft/server/EntityFishingHook.java
index 0fc1e85f..26b79b9f 100644
--- a/src/main/java/net/minecraft/server/EntityFishingHook.java
+++ b/src/main/java/net/minecraft/server/EntityFishingHook.java
@@ -183,6 +183,14 @@ public class EntityFishingHook extends Entity {
                 movingobjectposition = new MovingObjectPosition(entity);
             }
 
+            // PaperSpigot start - Allow fishing hooks to fly through players
+            if (movingobjectposition != null && movingobjectposition.entity instanceof EntityPlayer && owner != null && owner instanceof EntityPlayer) {
+                if (!((EntityPlayer) owner).getBukkitEntity().canSee(((EntityPlayer) movingobjectposition.entity).getBukkitEntity())) {
+                    movingobjectposition = null;
+                }
+            }
+            // PaperSpigot end
+
             if (movingobjectposition != null) {
                 org.bukkit.craftbukkit.event.CraftEventFactory.callProjectileHitEvent(this); // Craftbukkit - Call event
                 if (movingobjectposition.entity != null) {
-- 
2.20.1.windows.1

