From a2544a0029571b0fdbd570d9252f8ec9284be44b Mon Sep 17 00:00:00 2001
From: sathonay <sathonaypro@gmail.com>
Date: Fri, 30 Aug 2019 21:31:22 +0200
Subject: [PATCH] Clear arrows on player when he die


diff --git a/src/main/java/net/minecraft/server/EntityPlayer.java b/src/main/java/net/minecraft/server/EntityPlayer.java
index 6e9420f8..7cce9187 100644
--- a/src/main/java/net/minecraft/server/EntityPlayer.java
+++ b/src/main/java/net/minecraft/server/EntityPlayer.java
@@ -1161,6 +1161,7 @@ public class EntityPlayer extends EntityHuman implements ICrafting {
         this.lastDamager = null;
         this.combatTracker = new CombatTracker(this);
         this.lastSentExp = -1;
+        this.p(0);
         if (this.keepLevel || keepInventory) {
             this.exp = exp;
         } else {
-- 
2.20.1.windows.1

