From a302632584476f20377749c2cd027f32bd8c94d3 Mon Sep 17 00:00:00 2001
From: sathonay <sathonaypro@gmail.com>
Date: Fri, 30 Aug 2019 21:48:53 +0200
Subject: [PATCH] When player is teleported load the chunk if he is not load


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 826ea3c7..1604ad9d 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -519,6 +519,14 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
             getHandle().closeInventory();
         }
 
+        // nPaper start
+        // Check if player teleport location chunk is load , if it's not load it.
+        org.bukkit.Chunk chunk = to.getChunk();
+        if (!chunk.isLoaded()) {
+            chunk.load();
+        }
+        // nPaper end
+
         // Check if the fromWorld and toWorld are the same.
         if (fromWorld == toWorld) {
             entity.playerConnection.teleport(to);
-- 
2.20.1.windows.1

