From 984eb4138964b167f01e00a634fef0127959837d Mon Sep 17 00:00:00 2001
From: sathonay <sathonaypro@gmail.com>
Date: Fri, 30 Aug 2019 21:52:03 +0200
Subject: [PATCH] Fix head packet spam


diff --git a/src/main/java/net/minecraft/server/EntityTrackerEntry.java b/src/main/java/net/minecraft/server/EntityTrackerEntry.java
index d028f235..18735d1e 100644
--- a/src/main/java/net/minecraft/server/EntityTrackerEntry.java
+++ b/src/main/java/net/minecraft/server/EntityTrackerEntry.java
@@ -409,10 +409,14 @@ public class EntityTrackerEntry {
                         }
                     }
 
-                    // CraftBukkit start - Fix for nonsensical head yaw
-                    this.i = MathHelper.d(this.tracker.getHeadRotation() * 256.0F / 360.0F);
-                    this.broadcast(new PacketPlayOutEntityHeadRotation(this.tracker, (byte) i));
-                    // CraftBukkit end
+                    // nPaper start - fix head packet spam
+                    if (this.tracker instanceof EntityLiving) {
+                        // CraftBukkit start - Fix for nonsensical head yaw
+                        this.i = MathHelper.d(this.tracker.getHeadRotation() * 256.0F / 360.0F);
+                        this.broadcast(new PacketPlayOutEntityHeadRotation(this.tracker, (byte) i));
+                        // CraftBukkit end
+                    }
+                    // nPaper end
 
                     if (this.tracker instanceof EntityLiving) {
                         EntityLiving entityliving = (EntityLiving) this.tracker;
-- 
2.20.1.windows.1

